### ISSUE TITLE
Totally doesn't work on MSI GS65 Laptop

### ISSUE THREAD

**Issue by CryFrogg** [Askannz/optimus-manager#352]
**Describe the bug**
After configuring and installing all the requirements, I just can't have a desktop, simply "something went wrong and Manjaro can't recover...". 

**System info**
Please include :
- Manjaro GNOME 20.01
- GNOME
- GDM and installed gdm-prime as in the tutorial
- MSI GS65 with RTX 2080 maxq
- latest stable release (optimus-manager)
- have the video-hybrid-intel-nvidia-450xx-prime driver
**Logs**
`optimus-manager --status` gives me:
`ERROR: cannot check the active card (should be "intel"). Reason: Cannot run glxinfo : Failed to execute '__NV_PRIME_RENDE_OFFLOAD=0 glxinfo' : ERROR: unable to open display`
`Something went wrong during the liast GPU setup...`

Log file at /var/log/optimus-manager/switch/switch-....log/
```
[18] INFO: # Xorg pre-start hook
[18] INFO: Previous state was: {'type': 'pending_pre_xorg_start', 'requested_mode': 'intel', 'current_mode': None}
[18] INFO: Requested mode is: intel
[19] INFO: Checking for GDM display servers
[348] INFO: Available modules: ['nouveau', 'nvidia', 'nvidia_drm', 'nvidia_modeset', 'nvidia_uvm']
[348] INFO: Unloading modules ['nvidia_drm', 'nvidia_modeset', 'nvidia_uvm', 'nvidia'] (if loaded)
[373] INFO: Loaded extra Intel Xorg options (0 lines)
[374] INFO: Loaded extra Intel Xorg options (0 lines)
[374] INFO: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
[374] INFO: Writing state {'type': 'pending_post_xorg_start', 'switch_id': '20201211T200515', 'requested_mode': 'intel'}
[375] INFO: Xorg pre-start hook completed successfully.
[4] INFO: # Xorg post-start hook
[4] INFO: Running /etc/optimus-manager/xsetup-intel.sh
[11] INFO: Writing state {'type': 'done', 'switch_id': '20201211T200515', 'current_mode': 'intel'}
[11] INFO: Xorg post-start hook completed successfully.
```

Hope that I will get an solution, or else I can only use Bumblebee instead :)

**Comment by Hyper-KVM** [comment]
Do you have mesa-demos installed?

**Comment by CryFrogg** [comment]
Yes, I installed mesa-demos.

And I tried Manjaro GNOME 20.02 and the installation was fine,  and after reboot, I also get a desktop, which is already better.
But optimus-manager still failed working. This time however I have the `video-hybrid-intel-nvidia-455xx-prime` driver installed

After rebooting and command `optimus-manager --status` gives :
```
ERROR: a GPU setup was initiated but Xorg post-start hook did not run.
Log at /var/log/optimus-manager/switch/switch-20201211T233550.log
If your login manager is GDM, make sure to follow those instructions:
https://github.com/Askannz/optimus-manager#important--gnome-and-gdm-users
If your display manager is neither GDM, SDDM nor LightDM, or if you don't use one, read the wiki:
https://github.com/Askannz/optimus-manager/wiki/FAQ,-common-issues,-troubleshooting

Cannot execute command because of previous errors.
```
I already did all the works required by GNOME DE. E.g. install gdm-prime and so on.
The log file specified in the error message:
```
[21] INFO: # Xorg pre-start hook
[21] INFO: Previous state was: {'type': 'pending_pre_xorg_start', 'requested_mode': 'intel', 'current_mode': None}
[21] INFO: Requested mode is: intel
[22] INFO: Checking for GDM display servers
[300] INFO: Available modules: ['nouveau', 'bbswitch', 'acpi_call', 'nvidia', 'nvidia_drm', 'nvidia_modeset', 'nvidia_uvm']
[300] INFO: Unloading modules ['nvidia_drm', 'nvidia_modeset', 'nvidia_uvm', 'nvidia'] (if loaded)
[318] INFO: Loaded extra Intel Xorg options (0 lines)
[318] INFO: Loaded extra Intel Xorg options (0 lines)
[319] INFO: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
[319] INFO: Writing state {'type': 'pending_post_xorg_start', 'switch_id': '20201211T233550', 'requested_mode': 'intel'}
[319] INFO: Xorg pre-start hook completed successfully.
[4] INFO: # Xorg post-start hook
[4] INFO: Running /etc/optimus-manager/xsetup-intel.sh
[11] INFO: Writing state {'type': 'done', 'switch_id': '20201211T233550', 'current_mode': 'intel'}
[11] INFO: Xorg post-start hook completed successfully.
[7] INFO: # Xorg pre-start hook
[7] INFO: Previous state was: {'type': 'done', 'switch_id': '20201211T233550', 'current_mode': 'intel'}
[7] INFO: Requested mode is: intel
[7] INFO: Checking for GDM display servers
[1315] INFO: Loaded extra Intel Xorg options (0 lines)
[1315] INFO: Loaded extra Intel Xorg options (0 lines)
[1315] INFO: Writing to /etc/X11/xorg.conf.d/10-optimus-manager.conf
[1315] INFO: Writing state {'type': 'pending_post_xorg_start', 'switch_id': '20201211T233550', 'requested_mode': 'intel'}
[1315] INFO: Xorg pre-start hook completed successfully.
```

Another interesting thing, the file `90-mhwd.conf` in `/etc/X11/xorg.conf.d` does not contain anything, but the driver is indeed installed. Hope this will give a clue hmmm. Thanks btw

**Comment by CryFrogg** [comment]
The default graphics card should be intel. Buf after rebooting gain it uses automatically nvidia (because this laptop has a dedicated led indicator, I can tell which graphics card is in use.). And I don't know how to switch to intel T_T

**Comment by jdematte** [comment]
This is a problem with the gdm-prime package, i guess. I tried to use it with GNOME on my laptop and it simply doesn't work, but with any other desktop environment it does work as expected. #350 is another issue related to the gdm-prime package.

**Comment by CryFrogg** [comment]
is it possible then, that I just use the normal gdm instead of gdm-prime?

**Comment by CryFrogg** [comment]
And I also experienced the same issue after running the `prime-offload`, it says `Xorg post-start hook completed successfully`, but after switching the gpu, nothing has been changed.

**Comment by CryFrogg** [comment]
UPDATE : I downgrade the gdm-prime package to `gdm-prime 3.36.3-1` and it seems most of the things are working fine :). I appreciate your helps. Only after switching the graphics, I need to switch to a tty session and switch back which is annoying, but other things should be fine

**Comment by Askannz** [comment]
This was indeed a problem with GDM.
TL;DR, it now automatically close the Xorg server of the login screen, which caused a GPU switch to trigger again just after startup and put optimus-manager in an invalid state. Should be fixed now.
